go/pkg/repository/disk.go

```diff
@@ -150,6 +162,29 @@ func (s *DiskRepository) List(p string) ([]string, error) {
 	return result, nil
 }
 
+func (s *DiskRepository) ListTarFile(tarPath string) ([]string, error) {
+	fullTarPath := path.Join(s.rootDir, tarPath)
+	exists, err := files.FileExists(fullTarPath)
+	if err != nil {
+		return []string{}, err
+	}
+	if !exists {
+		return []string{}, &DoesNotExistError{msg: "ListTarFile: does not exist: " + fullTarPath}
```

Comment:
Since this message is shown to the end user, let's not include the internal function name.

Reply:
and `nil` instead of `[]string{}`

Reply:
Done for both.